### PR TITLE
Added MS Sculpt ISO-layout conversion with STM32F401

### PR DESCRIPTION
**Pull request by jn-bernard**
## Description

Added MS Sculpt ISO-layout conversion with STM32F401

## Types of Changes

A MS Sculpt ISO-layout conversion to QMK

### CODE REVIEW

**Comment by zvecr** [comment]
image

We wont be able to do anything with this PR in its current state. The history will need to be fixed so its just the intended keyboard addition.

**Comment by drashna** [comment]
Pretty sure this was meant to go to master, not a very old/outdated branch....

Changed to master

File: keyboards/ms_sculpt/config.h [hunk 1]
```diff
@@ -0,0 +1,37 @@
+// Copyright 2022 Jean Bernard (@jn-bernard)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+##pragma once
+
+#define RGB_DI_PIN A1
+#define RGBLED_NUM 16
+
+#define RGBLIGHT_ANIMATIONS
```
Comment (drashna) [inline keyboards/ms_sculpt/config.h:9]: Could you explicitly define which animations are enabled? 
```suggestion
#define RGBLIGHT_EFFECT_BREATHING
#define RGBLIGHT_EFFECT_RAINBOW_MOOD
#define RGBLIGHT_EFFECT_RAINBOW_SWIRL
#define RGBLIGHT_EFFECT_SNAKE
#define RGBLIGHT_EFFECT_KNIGHT
#define RGBLIGHT_EFFECT_CHRISTMAS
#define RGBLIGHT_EFFECT_STATIC_GRADIENT
#define RGBLIGHT_EFFECT_RGB_TEST
#define RGBLIGHT_EFFECT_ALTERNATING
#define RGBLIGHT_EFFECT_TWINKLE
```

File: keyboards/ms_sculpt/config.h [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+/*
+ * Feature disable options
+ *  These options are also useful to firmware size reduction.
+ */
+
+/* disable debug print */
+#define NO_DEBUG
+
+/* disable print */
+#define NO_PRINT
```
Comment (drashna) [inline keyboards/ms_sculpt/config.h:24]: If console isn't enabled, these are already defined.  If console is enabled, you probably want to have debug and print enabled, too.
```suggestion
```

File: keyboards/ms_sculpt/config.h [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+//#define DEBUG_MATRIX_SCAN_RATE
+#define BOOTMAGIC_LITE_ROW 4
+#define BOOTMAGIC_LITE_COLUMN 13
+
+/* disable action features */
+#define NO_ACTION_LAYER
+#define NO_ACTION_TAPPING
+#define NO_ACTION_ONESHOT
```
Comment (drashna) [inline keyboards/ms_sculpt/config.h:33]: Not sure you want these, as this disables layers, hold-tap keys, etc. 
```suggestion
/* disable action features */
// #define NO_ACTION_LAYER
// #define NO_ACTION_TAPPING
// #define NO_ACTION_ONESHOT
```

File: keyboards/ms_sculpt/halconf.h [hunk 4]
```diff
@@ -0,0 +1,8 @@
+#pragma once
```
Comment (drashna) [inline keyboards/ms_sculpt/halconf.h:1]: Could you add a GPL2+ compatible license header here? 

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

Or 

https://github.com/qmk/qmk_firmware/blob/6b299fab6e6a3143c061b9ea005253361fcd524b/data/templates/base/keyboard.c#L1-L15

File: keyboards/ms_sculpt/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,16 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/ms_sculpt/keymaps/default/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

Or 

https://github.com/qmk/qmk_firmware/blob/6b299fab6e6a3143c061b9ea005253361fcd524b/data/templates/base/keyboard.c#L1-L15

File: keyboards/ms_sculpt/keymaps/default/keymap.c [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+
+#define _BASE 0
```
Comment (drashna) [inline keyboards/ms_sculpt/keymaps/default/keymap.c:3]:
```suggestion
enum custom_layers {
    _BASE,
};
```

File: keyboards/ms_sculpt/rules.mk [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+DEBOUNCE = 3
+
+RGBLIGHT_ENABLE         = yes       # Enable global lighting effects. Do not enable with RGB Matrix
+RGBLIGHT_ANIMATIONS     = yes
+RGB_MATRIX_ENABLE       = no        # Enable per-key coordinate based RGB effects. Do not enable with RGBlight
+
+RGBLIGHT_FULL_POWER     = yes
+
+RAW_ENABLE = yes
+
```
Comment (drashna) [inline keyboards/ms_sculpt/rules.mk:12]: the rgblight values aren't used and should be removed.  The rest don't appear to be used. 
```suggestion
```

File: keyboards/ms_sculpt/sculpt_iso.h [hunk 8]
```diff
@@ -0,0 +1,22 @@
+#pragma once
```
Comment (drashna) [inline keyboards/ms_sculpt/sculpt_iso.h:1]: Could you add a GPL2+ compatible license header here? 

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

Or 

https://github.com/qmk/qmk_firmware/blob/6b299fab6e6a3143c061b9ea005253361fcd524b/data/templates/base/keyboard.c#L1-L15

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by jn-bernard — COMMENTED** [review]
fixed your issues

**Review by drashna — APPROVED** [review]

**Review by tzarc — APPROVED** [review]

**Pull request merged**